src/target_entry.rs

```diff
@@ -0,0 +1,104 @@
+use ffi;
+use glib;
+use glib::translate::*;
+use gobject_ffi;
+use std::mem;
+use TargetFlags;
+
+#[derive(Clone, Debug)]
+#[repr(C)]
+pub struct TargetEntry {
+    pub target: String,
+    pub flags: TargetFlags,
+    pub info: u32,
+}
+
+impl TargetEntry {
+    pub fn new(target: &str, flags: TargetFlags, info: u32) -> TargetEntry {
+        assert_initialized_main_thread!();
+        unsafe {
+            from_glib_full(ffi::gtk_target_entry_new(target.to_glib_none().0, flags.bits(), info))
+        }
+    }
+}
+
+#[doc(hidden)]
+impl Uninitialized for TargetEntry {
+    #[inline]
+    unsafe fn uninitialized() -> Self {
+        mem::uninitialized()
+    }
+}
+
+#[doc(hidden)]
+impl<'a> ToGlibPtr<'a, *const ffi::GtkTargetEntry> for TargetEntry {
+    type Storage = &'a Self;
+
+    #[inline]
+    fn to_glib_none(&'a self) -> Stash<'a, *const ffi::GtkTargetEntry, Self> {
+        let ptr: *const TargetEntry = &*self;
+        Stash(ptr as *const ffi::GtkTargetEntry, self)
```

Comment:
If I understand you correctly this should then be:
```
impl<'a> ToGlibPtrMut<'a, *mut ffi::GtkTargetEntry> for TargetEntry {
    type Storage = &'a mut Self;

    #[inline]
    fn to_glib_none_mut(&'a mut self) -> StashMut<'a, *mut ffi::GtkTargetEntry, (Self, ffi::GtkTargetEntry)> {
        let target_entry = ffi::GtkTargetEntry {
            target: self.as_ptr(),
            flags: self.flags.bits(),
            info: self.info,
        };
        StashMut((self, target_entry))
    }
}
```

But if I do that I get the following error:

```
$ cargo build
   Compiling gtk v0.1.1 (file:///home/susurrus/Projects/gtk)
error[E0277]: the trait bound `(target_entry::TargetEntry, ffi::GtkTargetEntry): glib::translate::ToGlibPtrMut<'a, *mut ffi::GtkTargetEntry>` is not satisfied
  --> src/target_entry.rs:52:5
   |
52 |       fn to_glib_none_mut(&'a mut self) -> StashMut<'a, *mut ffi::GtkTargetEntry, (Self, ffi::GtkTargetEntry)> {
   |  _____^ starting here...
53 | |         let target_entry = ffi::GtkTargetEntry {
54 | |             target: self.as_ptr(),
55 | |             flags: self.flags.bits(),
56 | |             info: self.info,
57 | |         };
58 | |         StashMut((self, target_entry))
59 | |     }
   | |_____^ ...ending here: the trait `glib::translate::ToGlibPtrMut<'a, *mut ffi::GtkTargetEntry>` is not implemented for `(target_entry::TargetEntry, ffi::GtkTargetEntry)`
   |
   = note: required by `glib::translate::StashMut`

error: aborting due to previous error

error: Could not compile `gtk`.

To learn more, run the command again with --verbose.
```

I should actually be implementing these traits on the type `(gtk::TargetEntry, ffi::GtkTargetEntry)`?